glossary (#999)
YosefLab/scvi-tools: docs/user_guide/developer.rst

   notebooks/module_user_guide
   notebooks/model_user_guide

Glossary
--------

.. tabs::

    .. tab:: Model

>         A model class is a user-facing object that contains the module as an attribute (i.e., ``self.module``).

Reviewer: ```suggestion
        A Model class inherits :class:`~scvi.model.base.BaseModelClass` and is the user-facing object for interacting with a Module.
```

>         The model has a `train` method that learns the parameters of the module, and also contains methods

Reviewer: ```suggestion
        The Model has a `train` method that learns the parameters of the Module, and also contains methods
```

>         The model has a `train` method that learns the parameters of the module, and also contains methods
>         for users to retrieve information from the module, like the latent representation of cells in a VAE.
>         Conventionally, the post-inference model methods should not store data into the AnnData object, but
>         instead return "standard" Python objects, like numpy arrays or pandas dataframes.
> 

Reviewer: ```suggestion
        The Model has a `train` method that learns the parameters of the Module, and also contains methods
        for users to retrieve information from the Module, like the latent representation of cells in a VAE.
        Conventionally, a Model's post-inference methods should not store data into the AnnData object, but
        instead return "standard" Python objects, like a numpy array or a pandas DataFrame.

```

    .. tab:: Module

>         A module is the lower-level object that defines a generative model and inference scheme. A module will

Reviewer: ```suggestion
        A Module is the lower-level object that defines a generative model and inference scheme. A Module will
```

        either inherit :class:`~scvi.module.base.BaseModuleClass` or :class:`~scvi.module.base.PyroBaseModuleClass`.

>         Consequently, a module can either be implemented with PyTorch alone, or Pyro. In the PyTorch only case, the

Reviewer: ```suggestion
        Consequently, a Module can either be implemented with PyTorch alone, or Pyro. In the PyTorch only case, the
```

        generative process and inference scheme are implemented respectively in the `generative` and `inference` methods,

>         while the `loss` method computes the, e.g, ELBO in the case of variational inference.

Reviewer: ```suggestion
        while the `loss` method computes the loss, e.g, ELBO in the case of variational inference.
```

.. tabs::

    .. tab:: TrainingPlan

>         The training plan is a PyTorch Lightning Module that is initializd with a scvi-tools module object.

Reviewer: ```suggestion
        The TrainingPlan is a PyTorch Lightning Module that is initializd with a scvi-tools module object.
```

        It configures the optimizers, defines the training step and validation step, and computes metrics to be
        recorded during training. The training step and validation step are functions that take data, run it through
        the model and return the loss, which will then be used to optimize the model parameters in the Trainer.

>         Overall, training plans can be used to develop complex inference schemes on top of modules.

Reviewer: ```suggestion
        Overall, child classes of TrainingPlan can be used to develop complex inference schemes on top of modules.
```
